src/main/java/org/apache/maven/shared/dependency/analyzer/ClassFileVisitorUtils.java

```diff
@@ -108,41 +106,30 @@ private static void acceptJar( URL url, ClassFileVisitor visitor )
     private static void acceptDirectory( File directory, ClassFileVisitor visitor )
         throws IOException
     {
-        if ( !directory.isDirectory() )
+    
+        List<Path> classFiles = Files.walk( directory.toPath() )
+            .filter( path -> path.toFile().getName().endsWith( ".class" ) )
```

Comment:
Why not `#getFileName()#toString()`?

Reply:
Six of one, half dozen of another. I looked at that and saw that getFileName returned a Path instead of a String, which seemed wrong to me. But if you prefer I can change this.

Reply:
Let's make consistently use NIO2.

Reply:
done

Reply:
might have to revert that. It seems to break the integration tests on Windows in a way that indicates a real issue.